src/main/java/cloud/fogbow/as/core/systemidp/DefaultSystemRolePlugin.java

```diff
@@ -0,0 +1,59 @@
+package cloud.fogbow.as.core.systemidp;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import cloud.fogbow.as.constants.ConfigurationPropertyKeys;
+import cloud.fogbow.as.constants.SystemConstants;
+import cloud.fogbow.as.core.PropertiesHolder;
+import cloud.fogbow.common.models.SystemUser;
+
+public class DefaultSystemRolePlugin implements SystemRolePlugin {
+
+	private Map<String, Set<String>> usersWithSpecialRoles;
+	private String defaultRole;
+	
+	public DefaultSystemRolePlugin() {
+		defaultRole = PropertiesHolder.getInstance().getProperty(ConfigurationPropertyKeys.DEFAULT_ROLE_KEY);
+		getUsersWithSpecialRoles();
+	}
+
+	private void getUsersWithSpecialRoles() {
+		usersWithSpecialRoles = new HashMap<String, Set<String>>();
+		String rolesNamesString = PropertiesHolder.getInstance().getProperty(ConfigurationPropertyKeys.ROLES_KEY);
+		
+		if (!rolesNamesString.isEmpty()) {
+			for (String roleName : rolesNamesString.trim().split(SystemConstants.ROLE_NAMES_SEPARATOR)) {
+				getUserNamesWithRole(roleName);
+			}			
+		}
+	}
+
+	private void getUserNamesWithRole(String roleName) {
+		String userNamesWithRoleString = PropertiesHolder.getInstance().getProperty(roleName);
+		
+		if (!userNamesWithRoleString.isEmpty()) {
+			for (String userName : userNamesWithRoleString.trim().split(SystemConstants.ROLE_NAMES_SEPARATOR)) {
+				if (!usersWithSpecialRoles.containsKey(userName)) {
+					usersWithSpecialRoles.put(userName, new HashSet<String>());
+				}
+				
+				usersWithSpecialRoles.get(userName).add(roleName);
+			}					
+		}
+	}
+	
+	public void setUserRoles(SystemUser user) {
+		String userName = user.getName();
```

Comment:
Here it should be id, instead of name, since id is guaranteed to be unique. The name field is only used in interfaces, providing a more friendly identification of the user. Nevertheless, in most cases the id is a friendly field as well. For instance, in LDAP, id is the usename used for login.

Reply:
Done.